src/components/Comment/index.spec.tsx

```diff
@@ -469,4 +468,93 @@ describe(__filename, () => {
       );
     });
   });
+
+  describe('discarding a new comment', () => {
+    it('renders a discard button to cancel entering a comment', () => {
+      const versionId = 33;
+      const fileName = 'file.name.json';
+      const line = 100;
+      const store = configureStore();
+      const payload = {
+        fileName,
+        line,
+        versionId,
+      };
+
+      store.dispatch(actions.beginComment(payload));
+      const dispatchSpy = spyOn(store, 'dispatch');
+      const root = render({
+        fileName,
+        line,
+        readOnly: false,
+        store,
+        versionId,
+      });
+      const discardButton = root.find(`.${styles.discardButton}`);
+      discardButton.simulate('click');
+
+      expect(discardButton).toHaveText('Discard');
+      expect(dispatchSpy).toHaveBeenCalledWith(
+        actions.considerDiscardComment(payload),
+      );
+    });
+
+    it('renders confirm button to discard a comment', () => {
+      const versionId = 33;
+      const fileName = 'file.name.json';
+      const line = 100;
+      const store = configureStore();
+      const payload = {
+        fileName,
+        line,
+        versionId,
+      };
+
+      store.dispatch(actions.beginComment(payload));
+      store.dispatch(actions.considerDiscardComment(payload));
+      const dispatchSpy = spyOn(store, 'dispatch');
+      const root = render({
+        fileName,
+        line,
+        readOnly: false,
+        store,
+        versionId,
+      });
+      const discardButton = root.find(`.${styles.discardButton}`);
+      discardButton.simulate('click');
+
+      expect(discardButton).toHaveText('Confirm discard');
+      expect(dispatchSpy).toHaveBeenCalledWith(actions.discardComment(payload));
+    });
+
+    it('provides a cancel button to abort discard', () => {
+      const versionId = 33;
+      const fileName = 'file.name.json';
+      const line = 100;
+      const store = configureStore();
+      const payload = {
+        fileName,
+        line,
+        versionId,
+      };
+
+      store.dispatch(actions.beginComment(payload));
+      store.dispatch(actions.considerDiscardComment(payload));
+      const dispatchSpy = spyOn(store, 'dispatch');
+      const root = render({
+        fileName,
+        line,
+        readOnly: false,
+        store,
+        versionId,
+      });
+      const abortDiscardButton = root.find(`.${styles.abortDiscardButton}`);
+      abortDiscardButton.simulate('click');
+
+      expect(abortDiscardButton).toHaveText('Cancel');
```

Comment:
Similar to my comment above, I think this assertion should appear before the call to `abortDiscardButton.simulate('click')` because you are actually checking the text *before* the button has been clicked. After the button has been clicked the button disappears.